p2p_wallet/Scenes/Main/ReceiveFundsViaLink/ClaimSentViaLinkTransaction.swift

```diff
@@ -20,9 +21,59 @@ struct ClaimSentViaLinkTransaction: RawTransactionType {
         Resolver.resolve(PricesServiceType.self).currentPrice(mint: token.address)?.value * tokenAmount
     }
     
-    let execution: () async throws -> TransactionID
-    
     func createRequest() async throws -> String {
-        try await execution()
+        // get receiver
+        guard let receiver = Resolver.resolve(UserWalletManager.self).wallet?.account.publicKey
+        else {
+            throw SolanaError.unauthorized
+        }
+        
+        // get services
+        let sendViaLinkDataService = Resolver.resolve(SendViaLinkDataService.self)
+        let contextManager = Resolver.resolve(RelayContextManager.self)
+        let solanaAPIClient = Resolver.resolve(SolanaAPIClient.self)
+        
+        let context = try await contextManager
```

Comment:
Get current context, but no update it!

Send via link screen is also confirm screen. Updating context will lead to inconsistent data.